Sources/SwiftSyntaxMacroExpansion/MacroSystem.swift

```diff
@@ -12,28 +12,337 @@
+
+  // The added attributes should be on their own line, so prepend them with a newline.
+  let indentedSource = "\n" + expanded.indented(by: member.indentationOfFirstLine)
+  return "\(raw: indentedSource)"
+}
+
+private func expandPeerMacroMember(
+  definition: PeerMacro.Type,
+  attributeNode: AttributeSyntax,
+  attachedTo: DeclSyntax,
+  in context: some MacroExpansionContext,
+  indentationWidth: Trivia
+) throws -> MemberDeclListSyntax? {
+  guard
+    let expanded = expandAttachedMacro(
+      definition: definition,
+      macroRole: .peer,
+      attributeNode: attributeNode.detach(in: context),
+      declarationNode: attachedTo.detach(in: context),
+      parentDeclNode: nil,
+      extendedType: nil,
+      conformanceList: nil,
+      in: context,
+      indentationWidth: indentationWidth
+    )
+  else {
+    return nil
+  }
+
+  // Separate the peers and the declaration by two newlines.
+  let indentedSource = "\n\n" + expanded.indented(by: attachedTo.indentationOfFirstLine)
+  return "\(raw: indentedSource)"
+}
+
+private func expandPeerMacroCodeItem(
+  definition: PeerMacro.Type,
+  attributeNode: AttributeSyntax,
+  attachedTo: DeclSyntax,
+  in context: some MacroExpansionContext,
+  indentationWidth: Trivia
+) throws -> CodeBlockItemListSyntax? {
+  guard
+    let expanded = expandAttachedMacro(
+      definition: definition,
+      macroRole: .peer,
+      attributeNode: attributeNode.detach(in: context),
+      declarationNode: attachedTo.detach(in: context),
+      parentDeclNode: nil,
+      extendedType: nil,
+      conformanceList: nil,
+      in: context,
+      indentationWidth: indentationWidth
+    )
+  else {
+    return nil
+  }
+
+  // Separate the peers and the declaration by two newlines.
+  let indentedSource = "\n\n" + expanded.indented(by: attachedTo.indentationOfFirstLine)
+  return "\(raw: indentedSource)"
+}
+
+/// Expand an accessor macro of a declaration that does not have existing
+/// accessors.
+///
+/// See comment in `expandAccessors` for an explanation why we need this.
+private func expandAccessorMacroWithoutExistingAccessors(
+  definition: AccessorMacro.Type,
+  attributeNode: AttributeSyntax,
+  attachedTo: DeclSyntax,
+  in context: some MacroExpansionContext,
+  indentationWidth: Trivia
+) throws -> AccessorBlockSyntax? {
+  guard
+    let expanded = expandAttachedMacro(
+      definition: definition,
+      macroRole: .accessor,
+      attributeNode: attributeNode.detach(in: context),
+      declarationNode: attachedTo.detach(in: context),
+      parentDeclNode: nil,
+      extendedType: nil,
+      conformanceList: nil,
+      in: context,
+      indentationWidth: indentationWidth
+    )
+  else {
+    return nil
+  }
+
+  // `expandAttachedMacro` adds the `{` and `}` to wrap the accessor block.
+  // Since the entire block is indented, we start it on a new line.
+  let indentedSource = "\n" + expanded.indented(by: attachedTo.indentationOfFirstLine)
+  return "\(raw: indentedSource)"
```

Comment:
Expand macros is just a snippet though, so that makes sense. `Inline Macro` would put it on the same line 🤷

My preference would be to trim + add a space, but I'm not going to block the PR on that.

Reply:
OK, changed it to start the accessor block on the same line.